src/rez/package_order.py

```diff
@@ -754,6 +754,104 @@ def from_pod(cls, data):
         return cls(packages=data["packages"])
 
 
+class PyPAPackageOrder(PackageOrder):
+    """A package order that allows for package ordering according to PEP440/PyPA.
+
+    A prerelease argument can be specified to order prerelease versions in front
+    of release versions
+
+    For example, given the versions [1.0b2, 1.0a1, 1.0, 1.1, 1.0rc1, 1.1b2],
+    an orderer initialized with ``prerelease=""`` would give the order:
+     [1.1, 1.1b2, 1.0, 1.0rc1, 1.0b2, 1.0a1].
+    an orderer initialized with ``prerelease="b"`` would give the order:
+     [1.1b2, 1.1, 1.0b2, 1.0rc1, 1.0, 1.0a1].
```

Comment:
the description above is out of date.  here's an attempt at a more complete conceptual description:

```
    Here's how this package orderer behaves: 
    1. it first separates versions into two groups based on the value of ``prerelease``:  a disallowed/non-preferential group and those which are allowed/preferred
    2. sorts each group of versions in the order defined by that spec.  e.g.  1.1 > 1.1rc1 > 1.1b1 > 1.1a1
    3. concatenates the two groups: placing the non-preferred group -- in sorted order -- after the preferred group.

    Note that since the package orderer does not have the ability to completely filter the non-preferred group 
    it could be possible for a resolve to result in a non-preferred version if all preferred versions are eliminated during
    resolution.

    With prerelease set to None, non-prerelease versions will be sorted before all prerelease versions. 
    With prerelease set to a prerelease string, order will prefer prerelease up to and including that level of risk tolerance (i.e. "beta", allows for "b" and "rc" releases) as well as non-prerelease versions. 

    For example, given the versions [1.0b2, 1.0a1, 1.0, 1.1, 1.0rc1, 1.1b2],
    an orderer initialized with ``prerelease=None`` would give the order:
     [1.1, 1.0, 1.1b2, 1.0rc1, 1.0b2, 1.0a1].
    an orderer initialized with ``prerelease="b"`` would give the order:
     [1.1, 1.1b2, 1.0, 1.0rc1, 1.0b2, 1.0a1].
```

Reply:
:+1: